docker_test.go

```diff
@@ -705,6 +705,40 @@ func Test_BuildContainerFromDockerfileWithBuildLog(t *testing.T) {
 	assert.Regexpf(t, `^Step\s*1/\d+\s*:\s*FROM alpine$`, temp[0], "Expected stdout first line to be %s. Got '%s'.", "Step 1/* : FROM alpine", temp[0])
 }
 
+func Test_BuildContainerFromDockerfileWithBuildLogWriter(t *testing.T) {
+	var buffer bytes.Buffer
+
+	ctx := context.Background()
+
+	// fromDockerfile {
+	req := ContainerRequest{
+		FromDockerfile: FromDockerfile{
+			Context:        filepath.Join(".", "testdata"),
+			Dockerfile:     "buildlog.Dockerfile",
+			PrintBuildLog:  true,
+			BuildLogWriter: &buffer,
+		},
+	}
+	// }
+
+	genContainerReq := GenericContainerRequest{
+		ProviderType:     providerType,
+		ContainerRequest: req,
+		Started:          true,
+	}
+
+	c, err := GenericContainer(ctx, genContainerReq)
+	CleanupContainer(t, c)
+	require.NoError(t, err)
+
+	out, err := io.ReadAll(&buffer)
+	require.NoError(t, err)
```

Comment:
suggestion: no need for this on a buffer you can get the string from `Bytes()` or `String()` depending on the output format needed.

Reply:
That's fair, I'll fix it